docs/docs/api/optimizers/GEPA.md

```diff
@@ -68,6 +68,318 @@ highest_score_achieved_per_task = new_prog.detailed_results.highest_score_achiev
+- **Domain-specific information**: Inject specialized knowledge, terminology, or context that the default proposer lacks
+- **Provider-specific prompting guides**: Optimize instructions for specific LLM providers (OpenAI, Anthropic, etc.) with their unique formatting preferences
+- **Coupled component updates**: Handle situations where 2 or more components need to be updated together in a coordinated manner, rather than optimizing each component independently
+- **External knowledge integration**: Connect to databases, APIs, or knowledge bases during instruction generation
+
+### Available Options
+
+**Built-in Options:**
+- **Default Proposer**: The standard GEPA instruction proposer (used when `instruction_proposer=None`). The default instruction proposer IS an instruction proposer as well! It is the most general one, that was used for the diverse experiments reported in the GEPA paper and tutorials.
+- **MultiModalInstructionProposer**: Handles `dspy.Image` inputs and structured multimodal content. This proposer has been specifically optimized for tasks that include one or more `dspy.Image` inputs.
+
+```python
+from dspy.teleprompt.gepa.instruction_proposal import MultiModalInstructionProposer
+
+# For tasks involving images or multimodal inputs
+gepa = dspy.GEPA(
+    metric=my_metric,
+    reflection_lm=dspy.LM(model="gpt-5", temperature=1.0, max_tokens=32000, api_key=api_key),
+    instruction_proposer=MultiModalInstructionProposer(),
+    auto="medium"
+)
+```
+
+We invite community contributions of new instruction proposers for specialized domains as the [GEPA library](https://github.com/gepa-ai/gepa) continues to grow. The GEPA repository contains multiple adapters (like `DefaultAdapter`, `DSPyFullProgramAdapter`, `TerminalBenchAdapter`, etc.) that, while not instruction proposers themselves, provide good examples and inspiration for implementing your own instruction proposer.
+
+### How to Implement Custom Proposer
+
+Custom instruction proposers must implement the `ProposalFn` protocol:
+
+```python
+def __call__(
+    self,
+    candidate: dict[str, str],           # Current component name -> instruction mapping
+    reflective_dataset: dict[str, list[dict[str, Any]]],  # Component -> failed examples  
+    components_to_update: list[str]      # Which components to improve
+) -> dict[str, str]:                     # Updated component name -> new instruction mapping
+```
+
+#### Basic Example: Word Limit Proposer
+
+```python
+import dspy
+from gepa.core.adapter import ProposalFn
+
+class GenerateWordLimitedInstruction(dspy.Signature):
+    """Given a current instruction and feedback examples, generate an improved instruction with word limit constraints."""
+    
+    current_instruction = dspy.InputField(desc="The current instruction that needs improvement")
+    issues_found = dspy.InputField(desc="Feedback and issues identified from failed examples")
```

Comment:
Similar, this isn't issues_found. There are 3 keys in "reflective_dataset" (Inputs, Generated Outputs, Feedback)

Reply:
👍